lang/funcs/encoding.go

```diff
@@ -26,7 +27,9 @@ var Base64DecodeFunc = function.New(&function.Spec{
 		if err != nil {
 			return cty.UnknownVal(cty.String), fmt.Errorf("failed to decode base64 data '%s'", s)
 		}
-
+		if !utf8.Valid([]byte(s)) {
+			return cty.UnknownVal(cty.String), fmt.Errorf("contents of '%s' are not valid UTF-8", s)
```

Comment:
Looking again at this, I'm having second thoughts about including the whole input string in the error message... it's likely to be a pretty long string of gibberish, and so maybe better to just say "the given string" in both cases here.

I'm torn, because this will make it harder to debug in cases where e.g. you accidentally pass something that _isn't_ base64, but our error message rendering code doesn't expect huge "words" in the text and so things will probably wrap badly in the output if a large base64 string appears there.